k8s minor bug fixes

willthames opened the pull request:
##### SUMMARY

Ensure `wait_condition`s with `Status: Unknown` actually
complete

Return k8s object after wait rather than k8s object before
wait when object is patched.



##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
module_utils/k8s/raw

##### ADDITIONAL INFORMATION
Candidate for 2.8 backport.

fabianvf commented on lib/ansible/module_utils/k8s/raw.py:
```diff
@@ -437,6 +436,9 @@ def _custom_condition(resource):
             if match.status == 'Unknown':
+                if match.status == condition['status']:
+                    if condition.get('reason'):
+                        return match.reason == condition['reason']
```
Should we also return one step up from this conditional, in case they didn't specify a `reason` but did specify a `status` of `Unknown`?

willthames replied:
Or change to `condition.get('reason', True)` ?

willthames replied:
Could also be more explicit with `condition.get('reason') or 'reason' not in condition`

willthames committed:
k8s minor bug fixes

Ensure `wait_condition`s with `Status: Unknown` actually
complete

Return k8s object after wait rather than k8s object before
wait when object is patched.

fabianvf commented:
shipit

willthames commented:
bot_status

resmo commented on lib/ansible/module_utils/k8s/raw.py:
```diff
@@ -347,7 +347,6 @@ def perform_action(self, resource, definition):
             if wait:
                 success, result['result'], result['duration'] = self.wait(resource, definition, wait_timeout, condition=wait_condition)
             match, diffs = self.diff_objects(existing.to_dict(), result['result'])
-            result['result'] = k8s_obj
```
is this removed by intention?

resmo replied:
ah I see duplicate code...never mind

resmo reviewed (approved):
LGTM
